Add EDITMODE_PRIORITY for ATLAS_TILE

guilhermefelipecgs opened the pull request:
Fix #24549.
Now it's possible to randomize ATLAS_TILE based on defined priority.
![peek 22-12-2018 21-48](https://user-images.githubusercontent.com/1387165/50379595-caae5700-0634-11e9-9fa5-ef2dc0681882.gif)

CowThing commented:
How does this work with `_forward_subtile_selection` in GDScript? It looks like there's now two methods with the same name but different arguments?

guilhermefelipecgs committed:
Add EDITMODE_PRIORITY for ATLAS_TILE

When editing ATLAS_TILE, now it's possible to enable priority to
randomize subtiles.

txigreman commented:
Maybe what I'm going to say it's stupid, but could "atlas tiles with priority" be implemented as autotiles with a 1x1 bitmask? Would that help to avoid having duplicated logic in different tile types?

guilhermefelipecgs commented:
Autotile with 1x1 bitmask doesn't make sense. AFAIK, the only way to get the same result using autotile is using the new wildcard feature #27565 with 3x3 bitmask and setting the center bitmask to on and all corners to wildcard.

> Would that help to avoid having duplicated logic in different tile types?

I don't understand why you think the logic is duplicated. Atlastile don't have bitmask, so, the code is a little different.

insomnius commented:
I really need this.

akien-mga commented:
Thanks!
